independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -795,13 +795,21 @@ private static void fetchType(Type type, BeanDeployment beanDeployment) {
     private static void collectCallbacks(ClassInfo clazz, List<MethodInfo> callbacks, DotName annotation, IndexView index,
             Set<String> knownMethods) {
         for (MethodInfo method : clazz.methods()) {
-            if (method.returnType().kind() == Kind.VOID && method.parameterTypes().isEmpty()) {
-                if (method.hasAnnotation(annotation) && !knownMethods.contains(method.name())) {
+            if (method.hasAnnotation(annotation) && !knownMethods.contains(method.name())) {
+                if (method.returnType().kind() == Kind.VOID && method.parameterTypes().isEmpty()) {
                     callbacks.add(method);
+                } else {
+                    // invalid signature - build a meaningful message.
+                    StringBuilder name = new StringBuilder(method.declaringClass() + "." + method.name() + "(");
+                    String params = method.parameters().stream().map(pi -> pi.type().name().toString())
```

Comment:
How about something like `method.parameterTypes().forEach(name::append)`? I know it's nitpicking... but this way we don't even need the lambda.

Reply:
It won't work; I need to join the parameter correctly (with `,`). 
I can use `String.join`, but that will require creating an intermediary list with the type names.

I've pushed another version using `MethodInfo.toString()`. Unfortunately, we lose navigability - the IDE won't open the class, but we avoid the lambda.

Reply:
Hm, and why not just `name.append(method.parametersTypes())`? That should be good enough, or?